UI/src/app/services/backend-connector.service.ts

```diff
@@ -26,27 +27,22 @@ export class BackendConnectorService {
     }
 
     private getData() {
+        
         return this.http
-            .get('localhost:3001/api')
-            .retry(this.connectionRetries)
-            .timeout(this.connectionTimeout)
-            .map((response: Response) =>
-                response.json() || {}
-            )
-            // .catch(this.handleError)
-            .subscribe(data => {
-                if(data) {
-                    const val = Data.parseData(data as JsonData);
-                    this.parseData(val.traces);
-                    this.parseLayout(val.layout);
-                    this.parseOptions(val.options);
+            .get('http://localhost:3001/api/nlptodata', 
+            {
+                params: {
+                    userquery: 'Plot number of employees for each sex as a pie chart for each departments',
+                     dataset: 'core_data'
```

Comment:
You want the name of the data set as a string here right?

Reply:
yes, I changed it here.